lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -989,6 +1013,10 @@ def __init__(
             mode (Mode): Whether to use unit or app certificates mode. Default is Mode.UNIT.
             refresh_events (List[BoundEvent]): A list of events to trigger a refresh of
               the certificates.
+            private_key (Optional[PrivateKey]): The private key to use for the certificates.
+                If provided, it will be used instead of generating a new one.
+                If the key is not valid the library will generate a new one.
+                Using this parameter is discouraged.
```

Comment:
Please explain why it is discouraged and the fact that not setting this parameter will mean that the private key will be automatically managed by the library.

Reply:
I'm not sure what would the best way to explain in a short and concise way to have it here.
I wrote `Using this parameter is discouraged, having to pass around private keys manually can be a security concern. Allowing the library to generate and manage the key is the more secure approach.`